### DISCUSSION TITLE
Why do function prototypes for C libraries have to be added manually?

### DISCUSSION THREAD

**Discussion by VisenDev** [vlang/v#18868] [Questions and Answers] [answered]
It seems odd that, in order to use a C library, you can `#include` the header file to get access to datatypes, however any functions from the library need to have their prototypes added manually (as far as I can tell). 

This seems odd as the entire point of header files is to avoid having to manually specify this. So why are these prototypes not just automatically converted to Vlang? Is there a technical issue preventing this or has this just not been implemented yet? 

Is this planned to be fixed in the future? If so, could someone show me where to go in the code to start looking into this?

This seems like a pretty major inconvenience to deal with when interfacing with C libraries, yet painless C interop seems to be one of the selling points of Vlang. 

Can anyone provide any details on this?

**Comment by JalonSolov** [comment, marked as answer]
No, you don't do the #include to get access to datatypes.  You do it so it can be passed on to the C compiler so that _it_ knows the types.

V doesn't have a C parser built in, so it can't read those files.  That's why you have to add V stucts, fn defs, etc. so it knows how to talk to the C code.

**Reply by VisenDev** [reply]
Thank you for the clarification, 
Do you know if there are plans to implement some sort of automatic translation to V for C header files? Similiar to Zig's `@cImport()`?

**Reply by JalonSolov** [reply]
It's in the 0.5 roadmap.
